tools/lint.sh

```
@@ -212,12 +212,402 @@ fi
+    echo "  - void __isr keyboard_irq_handler(void);"
+    echo "  - Helps compiler optimize for interrupt context"
+    echo ""
+    WARNINGS=$((WARNINGS + 1))
+else
+    echo -e "${GREEN}✓ IRQ handlers properly attributed${NC}"
+fi
+echo ""
+
+# Check 14: _Static_assert for Compile-Time Validation
+echo -e "${BLUE}[14/14] Checking for compile-time validation...${NC}"
+STATIC_ASSERT_COUNT=$(grep -R "${EXCLUDE_DIRS[@]}" --exclude="*.md" -c "_Static_assert\|#error" src/ 2>/dev/null | grep -v ":0$" | wc -l)
```

Shellcheck (0.11.0)

[style] Consider using 'grep -c' instead of 'grep|wc -l'.

(SC2126)